### PR TITLE
Bug fix/bts 722 (3.10)

### PR DESCRIPTION
**Pull request by hkernbach**
Backport of: https://github.com/arangodb/arangodb/pull/16766

NO EE companion.

### CODE REVIEW

File: arangod/Graph/PathType.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@ namespace graph {
 constexpr char const* KShortestPathsName = "K_SHORTEST_PATHS";
 constexpr char const* KPathsName = "K_PATHS";
 constexpr char const* AllShortestPathsName = "ALL_SHORTEST_PATHS";
+constexpr char const* ShortestPathName = "SHORTEST_PATH";
```
Comment (jsteemann) [inline arangod/Graph/PathType.cpp:36]: nit: the constant naming here is inconsistent.
According to our coding conventions, all names should  `k...Name`.
Apart from that, these `char const*`s should be `std::string_views`.
That doesn't need to be fixed in 3.10, but can be addressed in devel only.

File: arangod/Graph/PathType.h [hunk 2]
```diff
@@ -27,7 +27,12 @@ namespace arangodb {
 namespace graph {
 
 struct PathType {
-  enum class Type { KShortestPaths = 0, KPaths = 1, AllShortestPaths = 2 };
+  enum class Type {
+    KShortestPaths = 0,
+    KPaths = 1,
+    AllShortestPaths = 2,
+    ShortestPath = 3
```
Comment (jsteemann) [inline arangod/Graph/PathType.h:34]: Nit: according to our naming convention, all enum values should be `k...`.
This can be adjusted in devel only.

**Review by jsteemann — CHANGES_REQUESTED** [review]
LGTM. Would be nice if the code comments could be addressed in devel later.

**Review by jsteemann — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by KVS85** [comment]
@joerg84 will communicate https://github.com/arangodb/arangodb/pull/16894#pullrequestreview-1086869622 and https://github.com/arangodb/arangodb/pull/16894#pullrequestreview-1086868382 to @hkernbach.

**Pull request merged**